signalclient.go

```diff
@@ -81,26 +81,29 @@ func (c *SignalClient) Join(urlPrefix string, token string, params *ConnectParam
 	}
 
 	header := newHeaderWithToken(token)
-	conn, _, err := websocket.DefaultDialer.Dial(u.String(), header)
+	conn, hresp, err := websocket.DefaultDialer.Dial(u.String(), header)
 	if err != nil {
+		logger.Errorw("error establishing signal connection", err, "httpResponse", hresp)
 		// use validate endpoint to get the actual error
 		validateSuffix := strings.Replace(urlSuffix, "/rtc", "/rtc/validate", 1)
 
-		res, err := http.Get(ToHttpURL(urlPrefix) + validateSuffix)
-		if err != nil || res.StatusCode == http.StatusOK {
+		hresp, err := http.Get(ToHttpURL(urlPrefix) + validateSuffix)
+		if err != nil {
+			return nil, ErrCannotDialSignal
+		} else if hresp.StatusCode == http.StatusOK {
 			// no specific errors to return if validate succeeds
 			return nil, ErrCannotConnectSignal
```

Comment:
I am thinking this is the error for which the media self health should fail. This errors means, we were able to connect to the controller and do a validation. The validation was good, yet could not connect. That means, the media node cannot be picked/contacted.

Like I mentioned in the PR commit notes, I think if we can determine from WebSocket Dial return itself whether it is a dial failure or not, it would be great. Logging for now. We can potentially return errors during WebSocket dial and use those directly rather than second step.